src/Microsoft.Azure.WebJobs.Host/Bindings/BindingFactory.cs

```diff
@@ -115,7 +115,63 @@ public static class BindingFactory
             {
                 // Default, assume a Poco
                 return new PocoTriggerArgumentBinding<TMessage, TTriggerValue>(bindingStrategy, converterManager, elementType);
-            }         
+            }
+        }
+
+        /// <summary>
+        /// Creates a binding that binds to an <see cref="IAsyncCollector{T}"/> with the specified argument type. Allows for a binding
+        /// to be generated for any POCO type.
+        /// </summary>
+        /// <param name="parameter">The ParameterInfo for the binding</param>
+        /// <param name="collectorGenericType">The generic type that must implement <see cref="IAsyncCollector{T}"/> and have a public constructor with at most one parameter.</param>
+        /// <param name="collectorGenericArgumentType">The generic argument type for the <see cref="IAsyncCollector{T}"/></param>
+        /// <param name="converterManager">A converter manager to pass along.</param>        
+        /// <param name="invokeStringBinder">A <see cref="Func{T, TContext}"/> that returns the TContext to be used by the <see cref="IAsyncCollector{T}"/></param>
+        /// <returns></returns>
+        public static IBinding BindGenericCollector<TContext>(ParameterInfo parameter, Type collectorGenericType, Type collectorGenericArgumentType,
```

Comment:
It could -- but this method does nothing special other than pass the ConverterManager  along. 

The purpose of this is for when you want to support any kind of T in your IAsyncCollector<T> -- and you want the concrete Collector dynamically created based on the ParameterInfo. 

BindCollector requires you specify an builder Func that returns a concrete type. That requires a bit of reflection to handle -- so I moved all of that reflection in here. You tell this method the IAsyncCollector<T> type and the Type of T and it'll generate the builder Func for you (passing in a context object to the constructor). Everything else is passed along.

There may be a better way to do this internally in BindCollector if we refactor it a bit. We can do that in the future, I'd think.

---
In reply to: [55583043](https://github.com/Azure/azure-webjobs-sdk/pull/640#discussion_r55583043) [](ancestors = 55583043)